hopr/hopr-lib/src/testing/hopr.rs

```diff
@@ -25,7 +26,17 @@ impl TestedHopr {
         host_port: u16,
         db_path: String,
         safe: NodeSafeConfig,
+        auto_redeems: bool,
+        winn_prob: Option<f64>,
     ) -> Self {
+        #[cfg(test)]
+        {
+            // To properly run, tests rely on HOPR_TEST_DISABLE_CHECKS being set
+            if !std::env::var("HOPR_TEST_DISABLE_CHECKS").is_ok_and(|v| v.to_lowercase() == "true") {
+                panic!("HOPR_TEST_DISABLE_CHECKS envvar must be set for tests");
+            }
+        }
+
```

Comment:
I mean, the code that the env triggers, should be guided by a feature, not an env variable, if it is an undesirable behavior in production, it should be feature gated.